stravalib/model.py

```diff
@@ -149,58 +190,132 @@ def naive_datetime(value: Optional[AllDateTypes]) -> Optional[datetime]:
 class DeprecatedSerializableMixin(BaseModel):
     """
     Provides backward compatibility with legacy BaseEntity
+
+    Inherits from the `pydantic.BaseModel` class.
     """
 
+    # TODO: i used the output type from this class
+    # could Dict also be an int or float??
     @classmethod
-    def deserialize(cls, attribute_value_mapping: Dict):
+    def deserialize(
+        cls, attribute_value_mapping: Dict[str, str]
+    ) -> DeprecatedSerializableMixin:
         """
         Creates and returns a new object based on serialized (dict) struct.
+
+        Parameters
+        ----------
+        attribute_value_mapping : Dict
+            A dictionary representing the serialized data.
+
+        Returns
+        -------
+        DeprecatedSerializableMixin
+            A new instance of the class created from the serialized data.
+
+        Deprecated
+        ----------
+        1.0.0
+            The `deserialize()` method is deprecated in favor of `parse_obj()` method.
+            For more details, refer to the Pydantic documentation:
+            https://docs.pydantic.dev/usage/models/#helper-functions
+
+
         """
-        warn_method_deprecation(
+        exc.warn_method_deprecation(
```

Comment:
exc was imported above and then some modules were imported in a separate line. i just cleaned it up so we're calling exc. for each imported method, etc

Reply:
Looks good to me

Reply:
perfect!